regolith/mongoclient.py

```diff
@@ -337,8 +388,15 @@ def all_documents(self, collname, copy=True):
 
     def insert_one(self, dbname, collname, doc):
         """Inserts one document to a database/collection."""
+        doc = doc_cleanup(doc)
+        try:
+            valid = validate_doc(collname, doc, self.rc)
+            if not valid:
+                sys.exit("Validation failed. Upload cancelled... exiting program")
+        except Exception as e:
+            print(e)
+            sys.exit("Validation failed unexpectedly. Upload cancelled... exiting program")
```

Comment:
@sbillinge That makes sense. validate_doc does already print plenty of information, so I just removed the try/except and inserted a raise everywhere other than in insert_many, since I want it to insert the docs that don't fail validation.